### DISCUSSION TITLE
Limitations on number of Notebook Kernels

### DISCUSSION THREAD

**Discussion by gbrueckl** [microsoft/vscode-discussions#444] [Extension Development QnA] [answered]
Is there any known limitation of how many Notebook Kernels can be created using the Notebook API?
Or to be more precise, how many Kernels can exist at a given time?

it seems like we are facing an issue where Kernels get disposed again automatically without user interaction. Could there be another possible reason?

thanks in advance,
-gerhard

**Comment by DonJayamanne** [comment, marked as answer]
@gbrueckl 
There is no known limitation for the number of kernels that can be created

> it seems like we are facing an issue where Kernels get disposed again automatically without user interaction. Could there be another possible reason?

Sorry to hear you are running into this issue.

If this is related to the Jupyter extension, then this is an issue that used to exist in an older version of the extension and has since been resolved (please update to vscode 1.75 and the latest Jupyter extension). If that does not work, please do file an issue in the Jupyter repo.

If its not related to the Jupyter extension, please could you provide some more information on what extension you are using and hopefully provide some logs (details with the failure).

**Comment by gbrueckl** [comment]
any ideas what could cause the call to `.dispose()` on a notebook controller?
maybe when it throws an exception?
could it be related to outputs that are too large for renderers?

**Reply by DonJayamanne** [reply]
sorry, No idea, would need to see the logs
Please file an issue with the logs

here are instructions for the logs

* Open settings & go into `Jupyter->Logging`
* Change the value to `verbose`
* Reload VS Code,
* Attempt to repro the issue & then copy the output from the `Jupyter` output panel.
